soh/src/code/z_parameter.c

```diff
@@ -2529,6 +2529,21 @@ u16 Randomizer_Item_Give(PlayState* play, GetItemEntry giEntry) {
         return Return_Item_Entry(giEntry, RG_NONE);
     }
 
+    if (item == RG_TRIFORCE_PIECE) {
+        gSaveContext.triforcePiecesCollected++;
+        GameInteractor_SetTriforceHuntPieceGiven(true);
+
+        // Teleport to credits when goal is reached.
+        if (gSaveContext.triforcePiecesCollected == Randomizer_GetSettingValue(RSK_TRIFORCE_HUNT_PIECES_REQUIRED)) {
+            gSaveContext.sohStats.itemTimestamp[TIMESTAMP_TRIFORCE_COMPLETED] = GAMEPLAYSTAT_TOTAL_TIME;
+            gSaveContext.sohStats.gameComplete = 1;
+            Play_PerformSave(play);
+            GameInteractor_SetTriforceHuntCreditsWarpActive(true);
```

Comment:
is there a reason to do the gamecomplete and save here instead of in `GameInteractor`?

Reply:
The GameInteractor part only executes when gameplay isn't paused, while this code is ran when the textbox is still running. This ensures the time is set to when the player actually collects the last piece of the triforce and not when they also close the textbox, and allows people to reset the game during the textbox if they want to kill ganon afterwards, instead of being able to accidentily reset before the game autosaves.